leader-election-impl/src/test/java/com/palantir/paxos/SqlitePaxosStateLogTest.java

```diff
@@ -102,4 +127,34 @@ private static PaxosValue valueForRound(long round) {
         ThreadLocalRandom.current().nextBytes(bytes);
         return new PaxosValue("someLeader", round, bytes);
     }
+
+    private static Supplier<Connection> createReusableMemoizedConnection() {
+        Supplier<Connection> baseConnectionSupplier = SqliteConnections.createDatabaseForTest();
+        Supplier<Connection> nonClosingConnectionSupplier = bypassCloseOnConnection(baseConnectionSupplier);
+        return Suppliers.memoize(nonClosingConnectionSupplier::get);
+    }
+
+    private static Supplier<Connection> bypassCloseOnConnection(Supplier<Connection> connectionSupplier) {
+        return Suppliers.compose(conn ->
+                        (Connection) Proxy.newProxyInstance(Connection.class.getClassLoader(),
+                                new Class<?>[] {Connection.class},
+                                new CloseIgnoringInvocationHandler(conn)),
+                connectionSupplier::get);
+    }
+
+    private static final class CloseIgnoringInvocationHandler extends AbstractInvocationHandler {
```

Comment:
We talked offline so I know why this is needed, but probably best to document that.